homeworks/hw12/src/bank.rs

```diff
@@ -0,0 +1,869 @@
+                account: account.to_owned(),
+            }
+            .into())
+        } else {
+            let netxt_id = get_next_id();
+            self.accounts
+                .insert(account.to_owned(), RefCell::from(MONEY_ZERO));
+            let operation = Operation {
+                id: netxt_id.to_owned(),
+                source_account: account.to_owned(),
+                target_account: None,
+                amount: MONEY_ZERO,
+                operation_type: OperationType::CreateAccount,
+            };
+
+            //self.history.push_back(operation);
+            self.push_transaction(operation);
+            info!("Created account {}", &account);
+            Ok(netxt_id.clone().to_owned())
+        }
+    }
+    /// Deposits the specified amount into the account.
+    ///
+    /// # Arguments
+    ///
+    /// * `amount` - The amount to deposit into the account.
+    /// * `account` - The account to deposit into.
+    ///
+    /// Result
+    /// `TransactionId` for operation
+    /// # Errors
+    /// AmountNegativeError
+    /// AccountNotFoundError
+    ///
+    /// ```
+    fn deposit(&mut self, account: &str, amount: Money) -> Result<TransactionId, BankError> {
+        if let Some(balance) = self.accounts.get_mut(account) {
+            if amount <= Money::default() {
+                error!("Amount must be positive");
+                Err(AmountNegativeError {
+                    account: account.to_owned(),
+                    amount,
+                }
+                .into())
+            } else {
+                *balance.get_mut() += amount;
+                let transaction_id = get_next_id();
+                let operation = Operation {
+                    id: transaction_id.to_owned(),
+                    source_account: account.to_owned(),
+                    target_account: None,
+                    amount,
+                    operation_type: OperationType::Deposit,
+                };
+                // self.history.push_back(operation);
+                self.push_transaction(operation);
+                info!("Deposited into account {}", &account);
+                Ok(transaction_id.to_owned())
+            }
+        } else {
+            error!("Account does not exist");
+            Err(AccountNotFoundError {
+                account: account.to_owned(),
+            }
+            .into())
+        }
+    }
+    /// Withdraws the specified amount from the account.
+    ///
+    /// # Arguments
+    ///
+    /// * `amount` - The amount to withdraw from the account.
+    /// * `account` - The account to withdraw from.
+    ///
+    /// # Errors
+    /// AmountNegativeError
+    /// AccountNotFoundError
+    /// InsufficientFundsError
+    ///
+    /// Returns an error if the account balance is insufficient to cover the withdrawal amount.
+    ///
+    /// ```
+
+    fn withdraw(&mut self, account: &str, amount: Money) -> Result<TransactionId, BankError> {
```

Comment:
```suggestion
    /// ```
    fn withdraw(&mut self, account: &str, amount: Money) -> Result<TransactionId, BankError> {
```